token-metadata/program/src/processor.rs

```diff
@@ -1520,6 +1534,13 @@ pub fn set_and_verify_sized_collection_item(
     let mut metadata: Metadata = Metadata::from_account_info(metadata_info)?;
     let mut collection_metadata: Metadata = Metadata::from_account_info(collection_info)?;
 
+    // Don't reverify already verified items, otherwise we end up with invalid size data.
+    if let Some(ref collection) = metadata.collection {
```

Comment:
I think in this case it would be equivalent but I usually would do:
`if let Some(collection) = &metadata.collection` instead.

Do you have a reason to use `ref` keyword?
I don't know everything about this, but curious if you do know more about it.  Like is `ref` preferred.  I was thinking not needed because of this:
https://rust-lang.github.io/rfcs/2005-match-ergonomics.html

Reply:
In this case it's not actually necessary. I'm surprised `clippy` didn't catch it. For the other two functions thre needs to be a reference on one side or the other and it doesn't really matter but I suppose using `&` instead of `ref` is more idiomatic so I fixed that.